uaclient/files/state_files.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Dict
+
+from uaclient.data_types import BoolDataValue, DataObject, Field
+from uaclient.files.data_types import DataObjectFile
+from uaclient.files.files import UAFile
+
+SERVICES_ONCE_ENABLED = "services-once-enabled"
+
+
+class _ServicesOnceEnabledData(DataObject):
+    fields = [
+        Field("fips_updates", BoolDataValue, False),
+    ]
+
+    def __init__(self, fips_updates: bool):
+        self.fips_updates = fips_updates
+
+
+class _ServiceOnceEnableDataObjectFile(DataObjectFile):
+    def add_service(self, service: str):
+        file_content = self.read()
+        services_dict = {} if not file_content else file_content.to_dict()
+        service = service.replace("-", "_")
+
+        if service not in services_dict:
+            services_dict[service] = True
+            self.write(_ServicesOnceEnabledData(**services_dict))
+
+    def preprocess_data(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        # Since we are using now returning DataObject instances from read, we
+        # cannot have variables with "-" in them. We need to explictly convert
+        # them before creating the object
+        updated_data = {}
+        for key in data.keys():
+            if "-" in key:
+                updated_data[key.replace("-", "_")] = True
+
+        return updated_data
```

Comment:
If we're only supporting `fips_updates` as a field in this file then I'm not sure we need these to be so generic, but also I could see potential for needing a generic version in the future.

If we limit the functionality to what is strictly needed today we could do this:
```suggestion
    # def add_service(self, service: str):
    # not needed because we only have one service, so FipsUpdatesEntitlement can just do
    # services_once_enabled.fips_updates = True

    def preprocess_data(self, data: Dict[str, Any]) -> Dict[str, Any]:
        data["fips_updates"] = data.pop("fips-updates", None)
        return updated_data
```

But I could be convinced either way that the generic version is needed

Reply:
My logic here is exactly if we want to add more services in the future. I think it will not hurt if we keep it generic, but I can simplify it if needed. What do you think @renanrodrigo @CalvoM ?

Reply:
I'd say the `add_service` is also calling `write`, which would be needed direcly by the caller as well - so I like having it there.
About being generic, I think we can go either way. And since any service with two or more words in the name has the dash, this is good I guess.

Reply:
I am always for generic functions/methods.